gateway-js/src/index.ts

```diff
@@ -742,11 +817,11 @@ export class ApolloGateway implements GraphQLService {
       );
     }
 
-    return getServiceDefinitionsFromStorage({
+    return loadCsdlFromStorage({
```

Comment:
OK so just to be clear: When we publish this PR, upgrading will mean that you must get CSDL from the new pipeline? We are not having both managed federation mechanisms exist in parallel for any period of time?

I'm a little worried that this may be more challenging operationally.  This is a big PR and it would be good to actually merge it instead of keeping it over forever.  But the new pipeline may have some issues as it is first enabled. So ideally users would have a period where they could test the new pipeline but fall back to GCS if there are problems, and this process would not just be "downgrade `@apollo/gateway`" because there might be other improvements they would need. How hard would it be to do a release where both versions are supported and you opt in to the new thing with an ApolloGateway experimental constructor flag, and then a later release would change it to opt-out, and then later we delete the GCS code?

Reply:
> there might be other improvements they would need

If the reason we couldn't suggest them to downgrade would be because there are other improvements in the same version, could we instead just make sure to package this into its own release?

Reply:
Sure, but I mean, the process of stabilizing the new pipeline might take longer than a single release? I'm sure that the gateway code changes will be fine but switching from a years old pipeline to a brand new set of servers (whose fully production quality versions aren't ready today for sure) could take a little while.

Reply:
I checked in w/ the team implementing the Studio-side server and timelines suggest that we will be far happier if we can implement this as opt-in via `experimental_fetchCoreSchemasFromManagedFederation` or something rather than trying to coordinate this PR's merge and release with their schedules. (Which can be opt out in a later release and then removed after that.)

Reply:
Resolved in (a somewhat large) 307ee0eb